### PR TITLE
Add new `Product listing with gallery` pattern

### LINKED ISSUE #9110: Create `Product details: product listing with gallery and description` pattern
Product Listing with Gallery and Description (Wide)

Check Figma designs F5AAbRJOGbTeEweXPFtHgf-fi-1540_33725 for the latest updates and mobile views.

### PR DESCRIPTION
**Pull request by albarin**
This PR implements the `Product listing with gallery` pattern.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/9110

#### Design
Centered Search

#### User-Facing Testing

1. Create a new page or post
2. Make sure the `Product listing with gallery` pattern appears under the WooCommerce category dropdown.
3. Insert in and make sure it shows as expected on the design.

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

**Comment by albarin** [comment]
@shaunandrews this is the closest I could get to your design, what do you think we can do?
Also I think the same will happen with this issue https://github.com/woocommerce/woocommerce-blocks/issues/9112

**Comment by tomxygen** [comment]
I like the new pattern based on the figma design but I think the blockified product template needs some adjustments.
1. the  photo galley block is very limited, as the galley photos can't be aligned vertically like in the figma design.
2. quantity selector needs better styling, although in the figma design is not even there.
3. I would love to see the "save $x" block
4. the rating block could be improved by allowing to choose different colors for the stars and the text (yellow text doesnt really look good)

**Comment by shaunandrews** [comment]
Can we remove the quantity selector for the Add to Cart Button?

Can we improve the Ratings block to allow for customization of the text? Having it yellow makes it incredibly hard to read.

I'd love to have a _real_ gallery available, but until the we have the block in a better place, could we showcase additional images using the WP Core Image Gallery block?

Why is the price underlined? Is it a link or something?

**Comment by nefeline** [comment]
> Can we remove the quantity selector for the Add to Cart Button?

I was discussing this topic with @nerrad last week, and an alternative I'm exploring is to remove the restriction for the Add to Cart button block so it can be used within the Single Product block and not only as an inner block for the Products (Beta). This approach might be easier/faster than updating the Add to Cart From block to remove the quantity selector as that one is currently server-side rendered. Ref. pdnLyh-3xC-p2#comment-2443

**Comment by nefeline** [comment]
@albarin considering the Single Product block is experimental at this point: should we rely on such block (available only on dev builds) for creating patterns? What do you think?

**Comment by albarin** [comment]
> @albarin considering the Single Product block is experimental at this point: should we rely on such block (available only on dev builds) for creating patterns? What do you think?

Yes, I'll rebuild using normal blocks, not using the single product 👌

**Comment by roykho** [comment]
I am seeing errors in preview and after insertion.



Also I believe the title should be `Wicker Back Dining Chair...`? I see `Herriott 40" Solid Wood...`

**Comment by albarin** [comment]
@roykho Sorry, there was an issue with the images, updated the title as well. Thanks!

**Comment by albarin** [comment]
I made some changes to make it work better with differents widths

File: patterns/product-listing-with-gallery-and-description.php [hunk 1]
```diff
@@ -10,39 +10,39 @@
...
 			<!-- /wp:column -->
 
 			<!-- wp:column {"width":"66.66%","style":{"spacing":{"blockGap":"0","padding":{"right":"0","bottom":"0","left":"0","top":"20px"}}}} -->
 			<div class="wp-block-column" style="padding-top:20px;padding-right:0;padding-bottom:0;padding-left:0;flex-basis:66.66%">
-				<!-- wp:image {"width":380,"height":571,"sizeSlug":"full","linkDestination":"none"} -->
-				<figure class="wp-block-image size-full is-resized">
-					<img src="<?php echo esc_url( plugins_url( 'images/pattern-placeholders/product-furniture-4.png', dirname( __FILE__ ) ) ); ?>" alt="" width="380" height="571"/>
+				<!-- wp:image {"id":3869,"sizeSlug":"full","linkDestination":"none"} -->
+				<figure class="wp-block-image size-full">
+					<img src="http://store.local/wp-content/uploads/2023/05/product-furniture-4.png" alt="" class="wp-image-3869" />
```
Comment (roykho) [inline patterns/product-listing-with-gallery-and-description.php]: This link source is hardcoded.

**Comment by roykho** [comment]
Currently if you reduce the width (aka in mobile) of the browser, the gallery images are stacked on top of the main product image. Like this: 

What do you think about changing it so it looks like this: 

I was able to to do this by:
* On the columns that surrounds all images (turn off `Stack on mobile`).
* Go to each gallery image and click on `reset`.
* Go to the column that contains the gallery stack and turn on `Inner blocks use content width`. Set the width to 80px to more closely match the Figma design. Also set the `column settings width` to 80px. Go to `stylings` tab and give it a `padding-right` of 1 to again match the Figma design.

There is also an alignment issue when in full width with the description text:


That can be fixed by using `left justification`:

**Comment by tomxygen** [comment]
I tested this pattern included in the PR, however I'm not quite sure how to use it.
This pattern is not supposed to be added in the product template since it doesn't support dynamic content. 
It's basically like building a product page directly from the template.
Where am I supposed to use it? Should I create a new page and use it like a product page?
And how can I link a product in the catalog page to this new product page?

**Comment by albarin** [comment]
@roykho thanks for the review and instructions, I think I fixed everything you mentioned 🙏

**Comment by albarin** [comment]
> I tested this pattern included in the PR, however I'm not quite sure how to use it. This pattern is not supposed to be added in the product template since it doesn't support dynamic content. It's basically like building a product page directly from the template. Where am I supposed to use it? Should I create a new page and use it like a product page? And how can I link a product in the catalog page to this new product page?

@tomxygen thanks for your feedback 🙏 To answer your question I'm quoting @nefeline on this other pattern PR https://github.com/woocommerce/woocommerce-blocks/pull/9383:
>❗ Important note: This pattern was developed relying on WP core blocks. In this first moment, it serves the purpose of giving users ideas of how they can customize their stores rather than providing an out-of-the-box solution for store owners wishing to customize their single product templates and/or any posts/pages. As soon as we have the Single Product block available for general usage, this pattern, alongside others, will be updated to adopt it and, consequently, become more useful to the end user: more details available on https://github.com/woocommerce/woocommerce/issues/42374 .

**Review by roykho — APPROVED** [review]
Thanks for making the changes. In fullwidth there is no padding on the left/right side in mobile. But I don't think that is a blocker so approving for now.

**Pull request merged**